src/lib/helpers/package.ts

```diff
@@ -174,31 +175,31 @@ export const handleInstallation = async ({
   force,
   installDependency,
   dependencyDirPath,
-  dependencyTmpDirPath,
   explicitInstallation,
   name,
   version,
 }: HandleInstallationArg): Promise<void> => {
-  const installAndMoveToDependencyPath = async (): Promise<void> => {
-    await rm(dependencyTmpDirPath, { recursive: true, force: true });
-    await installDependency();
-    await rm(dependencyDirPath, { recursive: true, force: true });
-    await mkdir(dependencyDirPath, { recursive: true });
-    await rename(dependencyTmpDirPath, dependencyDirPath);
-  };
+  try {
+    // if dependency is already installed it will be there
+    await access(dependencyDirPath);
+  } catch {
+    if (!force) {
+      const isDependencyOverwrite = await confirm({
+        message: `Dependency ${name}@${version} already installed. Do you want to replace it?`,
+      });
 
-  if (force) {
-    await installAndMoveToDependencyPath();
-  } else {
-    try {
-      // if dependency is already installed it will be there
-      // so there is no need to install
-      await access(dependencyDirPath);
-    } catch {
-      await installAndMoveToDependencyPath();
+      if (!isDependencyOverwrite) {
+        commandObj.error(
+          `Cannot finish installation. Reason: dependency ${name}@${version} already installed`,
+        );
+      }
     }
+
+    await rm(dependencyDirPath, { recursive: true, force: true });
```

Comment:
don't really understand why you decided to make all these changes. Please explain. Why need to confirm stuff? Why you removed dependencyTmpDirPath (but for some reason not entirely as well, cause it still remains in type definition). I would not want to remove dependencyTmpDirPath cause I added that for the case when user starts installing something and doesn't wait till the end and excists the terminal. That happened several times and result is that some file is there but it didn't download completely so we it doesn't work (even though it can be accessed) So it's a bit more reliable to first download/install and then move file which is a very quick thing to do

Reply:
Regularly moving, force-recursive deleting and creating directories isn't very quick thing to do, it takes dozens of ms to do that.

Reply:
Nonetheless, i agree with your point. Will revert most of these changes.